x/ccv/provider/ibc_module.go

```diff
@@ -174,17 +174,25 @@ func (am AppModule) OnRecvPacket(
 ) ibcexported.Acknowledgement {
 	var (
 		ack            ibcexported.Acknowledgement
-		vscMaturedData ccv.VSCMaturedPacketData
-		slashData      ccv.SlashPacketData
+		consumerPacket ccv.ConsumerPacketData
 	)
-	if err := ccv.ModuleCdc.UnmarshalJSON(packet.GetData(), &vscMaturedData); err == nil {
+	// unmarshall consumer packet
+	if err := ccv.ModuleCdc.UnmarshalJSON(packet.GetData(), &consumerPacket); err != nil {
+		errAck := channeltypes.NewErrorAcknowledgement("cannot unmarshal CCV packet data")
+		ack = &errAck
+	}
+
+	// TODO: call ValidateBasic method on consumer packet data
```

Comment:
@MSalopek this is part of https://github.com/cosmos/interchain-security/issues/544